mongo/integration/mtest/mongotest.go

```diff
@@ -637,19 +637,30 @@ func (t *T) createTestClient() {
 	if clientOpts.Deployment == nil && t.clientType != Mock && clientOpts.ServerAPIOptions == nil && testContext.requireAPIVersion {
 		clientOpts.SetServerAPIOptions(options.ServerAPI(driver.TestServerAPIVersion))
 	}
-	// command monitor
+
+	// Setup command monitor
+	var customMonitor = clientOpts.Monitor
 	clientOpts.SetMonitor(&event.CommandMonitor{
```

Comment:
The `createTestClient` overwrites the monitor defined on the client options. We could treat the client options monitor as a callback in this overwriting.

Reply:
Could we just do
```
If clientOpts.Monitor != nil {
     clientOpts.SetMonitor(...)
}
```
Seems like the logic here is just to not overwrite the monitor if one is already defined on `t.clientOpts`.

Reply:
I think we still want mongotest to register the internal started events. Otherwise you wouldn't be able to run something like `(*T).ClearEvents` which depends on the `t.started = append(t.started, cse)` logic on line 650 (for instance).

You have this comment also:
>I believe mt.SetFailPoint will automatically disable the fail point with ClearFailPoints at the end of the test, so it seems like this defer is unnecessary.

This, for example, would not longer work if we set the custom monitor directly.

Reply:
Sounds good; per offline conversation, I didn't realize we need the existing monitor logic to append to the `started`, `failed` and `succeeded` slices.